### PR TITLE
Add custom VisualScript node tutorial

### PR DESCRIPTION
**Pull request by Jummit**

### CODE REVIEW

File: getting_started/scripting/visual_script/custom_nodes.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
+.. _doc_custom_nodes:
+
+Custom Nodes
+============
+
+Custom nodes are written in GDScript and can then be used in VisualScript.
+This is usefull for offloading complex code to GDScript and reusing it.
```
Comment (aaronfranke) [inline getting_started/scripting/visual_script/custom_nodes.rst]: Typo: `usefull` -> `useful`

File: getting_started/scripting/visual_script/custom_nodes.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+There are some functions that can be implemented to set parameters of the custom node.
+Only add functions that are needed, a ``_has_input_sequence_port`` function is not necessary if it should return ``false`` for example.
+
+The most important part of a custom node is the ``_step`` function. The logic of the node is defined there.
+
+The ``inputs`` parameter holds the value of the input ports.
+
+The ``outputs`` parameter is an array where the indices represent the output port ids. It can be modified to set the values of the output ports.
+
+``start_mode`` can be check to see if it is the first time ``_step`` is called.
```
Comment (aaronfranke) [inline getting_started/scripting/visual_script/custom_nodes.rst]: `check` -> `checked`

**Comment by pycbouh** [comment]
This seems similar to [the article on custom VisualShader nodes](https://docs.godotengine.org/en/stable/tutorials/plugins/editor/visual_shader_plugins.html), which is placed in *Editor plugins* section for whatever reason.

We should probably either put this article alongside it, or move the shader article to a more appropriate place, like https://docs.godotengine.org/en/stable/tutorials/shading/visual_shaders.html.

**Comment by Calinou** [comment]
@pycbouh I think moving the shader article to the Shading section would make more sense, so let's merge this for now.

File: getting_started/scripting/visual_script/index.rst [hunk 3]
```diff
@@ -8,4 +8,5 @@ VisualScript
    what_is_visual_scripting
    getting_started
    nodes_purposes
+   custom_nodes
```
Comment (Calinou) [inline getting_started/scripting/visual_script/index.rst:11]:
```suggestion
   custom_visualscript_nodes
```

File: getting_started/scripting/visual_script/custom_nodes.rst [hunk 4]
```diff
@@ -0,0 +1,102 @@
+.. _doc_custom_nodes:
+
+Custom Nodes
```
Comment (Calinou) [inline getting_started/scripting/visual_script/custom_nodes.rst:3]:
```suggestion
Custom VisualScript Nodes
```

File: getting_started/scripting/visual_script/custom_nodes.rst [hunk 5]
```diff
@@ -0,0 +1,102 @@
+.. _doc_custom_nodes:
```
Comment (Calinou) [inline getting_started/scripting/visual_script/custom_nodes.rst:1]:
```suggestion
.. _doc_custom_visualscript_nodes:
```

File: getting_started/scripting/visual_script/custom_nodes.rst [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+Custom Nodes
+============
```
Comment (Calinou) [inline getting_started/scripting/visual_script/custom_nodes.rst:4]:
```suggestion
=========================
```

**Comment by Jummit** [comment]
> Sorry for the late review.

No problem, this is a very niche tutorial. I renamed what you suggested.

File: getting_started/scripting/visual_script/index.rst [hunk 7]
```diff
@@ -8,4 +8,4 @@ VisualScript
    what_is_visual_scripting
    getting_started
    nodes_purposes
-
+   custom_visualscript_nodes
```
Comment (Calinou) [inline getting_started/scripting/visual_script/index.rst:11]: Remove the blank line above `custom_visualscript_nodes`.

Comment (Jummit) [reply getting_started/scripting/visual_script/index.rst:11]: I think this is fixed now.

**Comment by Calinou** [comment]
Thanks!

**Pull request merged**